include/itkVkDiscreteGaussianImageFilter.hxx

```diff
@@ -0,0 +1,181 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkVkDiscreteGaussianImageFilter_hxx
+#define itkVkDiscreteGaussianImageFilter_hxx
+
+#include "itkFFTDiscreteGaussianImageFilter.h"
+#include "itkDiscreteGaussianImageFilter.h"
+#include "itkImageToImageFilter.h"
+#include "itkVkBlurringPerformanceMetric.h"
+
+namespace itk
+{
+template <typename TInputImage, typename TOutputImage>
+void
+VkDiscreteGaussianImageFilter<TInputImage, TOutputImage>::GenerateInputRequestedRegion()
+{
+  ImageToImageFilter<TInputImage, TOutputImage>::GenerateInputRequestedRegion();
+  auto inputRegion = this->GetOutput()->GetRequestedRegion();
+
+  // Pad the requested region by the kernel radius.
+  inputRegion.PadByRadius(this->GetKernelRadius());
+
+  // Crop the output requested region to fit within the largest
+  // possible region.
+  InputImageType * inputPtr = itkDynamicCastInDebugMode<InputImageType *>(this->GetPrimaryInput());
+  bool             wasPartiallyInside = inputRegion.Crop(inputPtr->GetLargestPossibleRegion());
+  if (!wasPartiallyInside)
+  {
+    itkExceptionMacro("Requested region is outside the largest possible region.")
+  }
+
+  // Input is an image, cast away the constness so we can set
+  // the requested region.
+  inputPtr->SetRequestedRegion(inputRegion);
```

Comment:
The "constness" part of the comment appears to be out of date.

Reply:
Thank you for the fast review!

Comment removed.